build/nuget-release.yml

```diff
@@ -0,0 +1,163 @@
+          - task: DotNetCoreInstaller@0
+            displayName: 'Import .NET Core SDK ($(DotNet.Sdk.Version))'
+            inputs:
+              version: '$(DotNet.Sdk.Version)'
+          - task: qetza.replacetokens.replacetokens-task.replacetokens@3
+            displayName: 'Replace integration test tokens'
+            inputs:
+              rootDirectory: 'src/$(Project).Tests.Integration/'
+              targetFiles: 'appsettings.json'
+              encoding: 'auto'
+              writeBOM: true
+              actionOnMissing: 'fail'
+              keepToken: false
+              tokenPrefix: '#{'
+              tokenSuffix: '}#'
+          - task: DotNetCoreCLI@2
+            displayName: 'Run integration tests'
+            inputs:
+              command: test
+              projects: 'src/**/$(Project).Tests.Integration.csproj'
+              arguments: '--configuration $(Build.Configuration)'
+              nobuild: true
+              publishTestResults: true
+
+  - stage: Release
+    displayName: 'Release'
+    dependsOn:
+      - UnitTests
+      - IntegrationTests
+    condition: succeeded()
+    jobs:
+      - job: PushRelease
+        displayName: 'Push release'
+        pool:
+          vmImage: 'ubuntu-16.04'
+        steps:
+          - task: DownloadPipelineArtifact@2
+            displayName: 'Download build artifacts'
+            inputs:
+              artifact: 'Build'
+              path: '$(Build.SourcesDirectory)'
+          - task: GitHubRelease@0
+            displayName: 'Create GitHub Release (Conditional Step)'
+            inputs:
+              gitHubConnection: 'GitHub (arcus-automation - OAuth)'
+              repositoryName: 'arcus-azure/arcus.eventgrid'
+              tagSource: manual
+              tag: 'v$(Build.BuildNumber)'
+              title: 'v$(Build.BuildNumber)'
+              releaseNotesSource: input
+              releaseNotes: |
+                Install new version via [NuGet](https://www.nuget.org/packages/Arcus.EventGrid.All/$(Build.BuildNumber))
+                ```shell
+                > Install-Package Arcus.EventGrid.All -Version $(Build.BuildNumber)
+                ```
+            condition: not(contains(variables['Build.BuildNumber'], '-'))
+          - task: GitHubRelease@0
+            displayName: 'Create GitHub Pre-Release (Conditional Step)'
+            inputs:
+              gitHubConnection: 'GitHub (arcus-automation - OAuth)'
+              repositoryName: 'arcus-azure/arcus.eventgrid'
+              tagSource: manual
+              tag: 'v$(Build.BuildNumber)'
+              title: 'v$(Build.BuildNumber)'
+              releaseNotesSource: input
+              releaseNotes: |
+                Install new version via [NuGet](https://www.nuget.org/packages/Arcus.EventGrid.All/$(Build.BuildNumber))
+                ```shell
+                > Install-Package Arcus.EventGrid.All -Version $(Build.BuildNumber)
+                ```
+              isPreRelease: true
```

Comment:
I see that there is a `compareWith: 'lastRelease'` which I would use for preview packages. It limits the release notes for this release to only the delta of the last non-draft which is what we want I think?

Otherwise if we create 4 previews, preview 4 will include changes from 1, 2 & 3 which makes less sense, no?

Reply:
👍